### PR TITLE
Add support for Razer Cobra

### LINKED ISSUE #2145: Support for Razer Cobra
### Device Info

* Name: Razer Cobra
* Product Number: RZ01-04650100
* Model Number: RZ01-0465

### Device Information
```
Please post the output of
    lsusb -d 1532: -v
```
```
Bus 002 Device 002: ID 1532:00a3 Razer USA, Ltd Razer Cobra
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x00a3 
  bcdDevice            2.00
  iManufacturer           1 Razer
  iProduct                2 Razer Cobra
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     159
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0010  1x 16 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
Device Status:     0x0000
  (Bus Powered)
```
```
Please post the output of
    ls -lah /dev/input/by-id/
```
```
итого 0
drwxr-xr-x 2 root root 140 окт 18 12:55 .
drwxr-xr-x 4 root root 460 окт 18 12:55 ..
lrwxrwxrwx 1 root root   9 окт 18 12:55 usb-Razer_Razer_Cobra-event-if01 -> ../event4
lrwxrwxrwx 1 root root   9 окт 18 12:55 usb-Razer_Razer_Cobra-event-mouse -> ../event5
lrwxrwxrwx 1 root root   9 окт 18 12:55 usb-Razer_Razer_Cobra-if01-event-kbd -> ../event3
lrwxrwxrwx 1 root root   9 окт 18 12:55 usb-Razer_Razer_Cobra-if02-event-kbd -> ../event2
lrwxrwxrwx 1 root root   9 окт 18 12:55 usb-Razer_Razer_Cobra-mouse -> ../mouse0
```
```
Please post the output of
    usbhid-dump -m 1532 -ed
```
```
002:002:002:DESCRIPTOR         1697621467.288601
 05 01 09 06 A1 01 05 07 19 E0 29 E7 15 00 25 01
 75 01 95 08 81 02 81 01 19 00 2A FF 00 15 00 26
 FF 00 75 08 95 06 81 00 05 08 19 01 29 03 15 00
 25 01 75 01 95 03 91 02 95 05 91 01 C0

002:002:001:DESCRIPTOR         1697621467.290842
 05 01 09 06 A1 01 85 01 05 07 19 E0 29 E7 15 00
 25 01 75 01 95 08 81 02 19 00 2A FF 00 15 00 26
 FF 00 75 08 95 0E 81 00 C0 05 0C 09 01 A1 01 85
 02 19 00 2A 3C 02 15 00 26 3C 02 95 01 75 10 81
 00 75 08 95 0D 81 01 C0 05 01 09 80 A1 01 85 03
 19 81 29 83 15 00 25 01 75 01 95 03 81 02 95 05
 81 01 75 08 95 0E 81 01 C0 05 01 09 00 A1 01 85
 04 09 03 15 00 26 FF 00 35 00 46 FF 00 75 08 95
 0F 81 00 C0 05 01 09 00 A1 01 85 05 09 03 15 00
 26 FF 00 35 00 46 FF 00 75 08 95 0F 81 00 C0

002:002:000:DESCRIPTOR         1697621467.293723
 05 01 09 02 A1 01 09 01 A1 00 05 09 19 01 29 05
 15 00 25 01 75 01 95 05 81 02 75 01 95 03 81 03
 06 00 FF 09 40 75 08 95 02 15 81 25 7F 81 02 05
 01 09 38 15 81 25 7F 75 08 95 01 81 06 09 30 09
 31 16 00 80 26 FF 7F 75 10 95 02 81 06 C0 06 00
 FF 09 02 15 00 25 01 75 08 95 5A B1 01 C0

```
### Packet Captures

Cannot do it now. I hope it isn't necessary.

### Synapse Screenshots
Image 015
Image 017
Image 018
Image 024

### PR DESCRIPTION
**Pull request by hitchho**
Fixes #2145

### CODE REVIEW

File: daemon/openrazer_daemon/hardware/mouse.py [hunk 1]
```diff
@@ -1719,11 +1719,47 @@ def _close(self):
...
+
+        self._battery_manager.close()
+
+
 class RazerViperV2ProWired(__RazerDevice):
     """
     Class for the Razer Viper V2 Pro (Wired)
     """
-    EVENT_FILE_REGEX = re.compile(r'.*usb-Razer_Razer_Viper_V2_Pro_000000000000-if0(1|2)-event-kbd')
+    EVENT_FILE_REGEX = re.compile(r'.*usb-Razer_Razer_Viper_V2_Pro-000000000000-if0(1|2)-event-kbd')
```
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/mouse.py]: Why did you modify this device?

**Comment by z3ntu** [comment]
Please check the CI failure :)

File: daemon/openrazer_daemon/hardware/mouse.py (hunk 1 above)
Comment (hitchho) [reply daemon/openrazer_daemon/hardware/mouse.py]: by accident edited wrong line, tried to modify it back to exact same line as it was but seems to still differ

**Comment by hitchho** [comment]
gave test run for these and not a single button or mouse movement worked, so still in progress something clearly missing

**Comment by z3ntu** [comment]
Not sure how exactly this can happen, normally all the input handling is still handled by the core kernel, and only a few extras as most are provided by the razermouse driver. At most I could imagine you could try touching `expected_subclass` for your device, but I think this should not be responsible for what you're apparently seeing.

**Pull request closed without merging**